webui/react/src/pages/Admin/GroupManagement.tsx

```diff
@@ -135,6 +141,12 @@ const GroupManagement: React.FC = () => {
 
   const fetchGroups = useCallback(async (): Promise<void> => {
     if (!('tableLimit' in settings) || !('tableOffset' in settings)) return;
+    if (settings.tableOffset > 0 && settings.tableOffset === total) {
+      // handle when last group is deleted on a page
+      updateSettings({
+        tableOffset: settings.tableOffset - settings.tableLimit,
+      });
+    }
```

Comment:
Does this work in practice? I would assume that `updateSettings` would set the table offset, but `settings` wouldn't update until the next render, so you'd get the same last page in this call.

is there a way to do this that doesn't fire a superfluous `getGroups` call?

Reply:
returning out of the fetch function from this conditional seems to avoid the extra call